Sage/carbon
# fix(tab-title): remove additional error border and prevent content jumping when has siblings FE-4943

fix #4833
fix #4875

### Proposed behaviour



Updates `TabTitle` styling to remove additional right border when error state detected
![image](https://user-images.githubusercontent.com/44157880/154990807-9935be48-a313-4743-8e0e-265885703e7d.png)

Prevents content jumping when `siblings` prop used and `isTabSelected`.
![image](https://user-images.githubusercontent.com/44157880/154990966-e3c06792-f532-4d3f-9958-c3c193a87b85.png)

Updates styling to remove min-width when `customLayout` is used and updates demo to use `Box` and `flex` layout.
![image](https://user-images.githubusercontent.com/44157880/154991043-e2fd9f8a-26d9-40d5-90f9-d54970341f75.png)

### Current behaviour

Extra border-right is added when error state and alternateStyling is used (alternate-styling story).
![image](https://user-images.githubusercontent.com/44157880/154990389-2046bc57-36dd-4e58-af5a-ebae7d56723c.png)

The content jumps when a tab with sibling elements is selected.
![image](https://user-images.githubusercontent.com/44157880/154990596-80721913-9c71-472e-9cc5-380d34a4a9c7.png)

The `customLayout` story uses grid layout and sets the container `min-width: 100px` meaning it overflows the `TabTitle` when error state is active.
![image](https://user-images.githubusercontent.com/44157880/154990718-454dabb0-f9cd-416d-94a0-370f5b1dada7.png)




### Checklist




#### QA


### Additional context



### Testing instructions


Check Tab stories on storybook

The following CodeSandbox is an example of the broken behaviour.
You can see the new behaviour by looking at the version in the comment by `codesandbox[bot]`.
